array/unary.gen.go

```diff
@@ -0,0 +1,139 @@
+// Generated by tmpl
+// https://github.com/benbjohnson/tmpl
+//
+// DO NOT EDIT!
+// Source: unary.gen.go.tmpl
+
+package array
+
+import (
+	"github.com/apache/arrow/go/v7/arrow/memory"
+)
+
+//
+// Arithmetic Ops
+//
+
+func IntUnarySub(v *Int, mem memory.Allocator) (*Int, error) {
+	n := v.Len()
+	b := NewIntBuilder(mem)
+	b.Resize(n)
+	for i := 0; i < n; i++ {
+		if v.IsValid(i) {
+			b.Append(-v.Value(i))
```

Comment:
I was curious what Go does if you try to negate `math.MinInt64`:
https://go.dev/play/p/h-QiLnQU8nz

I guess we inherit this behavior...